### ISSUE TITLE
[macOS] Support --obfuscate flag for macOS builds

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#131499]
Support [Dart obfuscation](https://docs.flutter.dev/deployment/obfuscate) for macOS in the tool the same way we do for iOS and Android.

Extracted from request here: https://github.com/flutter/flutter/issues/71674#issuecomment-1647826652

**Comment by jonahwilliams** [comment]
This is already supported?

**Comment by cbracken** [comment]
@jonahwilliams you're right this is a dupe of flutter/website#9046 (filed quickly as we were blasting through triage), the issue is really just that we don't document the steps in a way that's clear for macOS builds.
